helix-core/src/test/java/org/apache/helix/manager/zk/TestZNRecordSizeLimit.java

```diff
@@ -86,8 +89,8 @@ public void testZNRecordSizeLimitUseZNRecordSerializer() {
     _gZkClient.createPersistent(path2, true);
     try {
       _gZkClient.writeData(path2, largeRecord);
-    } catch (HelixException e) {
-      Assert.fail("Should not fail because data size is larger than 1M since compression applied");
+    } catch (ZkMarshallingError e) {
```

Comment:
What is this test log testing?

If the expectation is no error, then we don't care what is the exception. If the operation fails, just throw it. If we use Assert.fail(), we get a message, but we are actually not sure the detail of the exception, because we never include it in the assert fail message.
If the expectation is we have an error, then the assert in the catch block will fail the test.

Can we make it straight?

Reply:
We are testing a `ZkMarshallingError` should not be thrown by the serializer. This message specifies the reason why this bad exception is thrown.
FYI, exception stack trace is added to the message.

Reply:
I don't have a strong preference here. But I think it is redundant try-catch. Otherwise, why we are not adding this try-catch then throw exception blocks to all the test code? Because all test code, as long as it is not testing for the negative cases, will have the same assumption that we shall not have a certain Exception thrown.

Reply:
I understand your point. And I agree we shouldn’t do redundant code. But in this specific case here, since compression is such a critical part, I would prefer to make it bold. If there is anything change in the logic and triggers the test failed, the message should tell the reason. And one more reason I use assert.fail is to avoid that if someone changes the test and catch the exception Without failing the test , the exception thrown from here may not fail the test. This is what I would like to prevent.